e2e/terraform/README.md

```diff
@@ -37,6 +33,67 @@ cd e2e/terraform/
 terraform apply
 ```
 
+> Note: You will likely see "Connection refused" or "Permission denied" errors
+> in the logs as the provisioning script run by Terraform hits an instance
+> where the ssh service isn't yet ready. That's ok and expected; they'll get
+> retried. In particular, Windows instances can take a few minutes before ssh
+> is ready.
+
+## Nomad Version
+
+You'll need to pass one of the following variables in either your
+`terraform.tfvars` file or as a command line argument (ex. `terraform apply
+-var 'nomad_version=0.10.2+ent'`)
+
+* `nomad_local_binary`: provision this specific local binary of Nomad. This is
+  a path to a Nomad binary on your own host. Ex. `nomad_local_binary =
+  "/home/me/nomad"`.
+* `nomad_sha`: provision this specific sha from S3. This is a Nomad binary
+  identified by its full commit SHA that's stored in a shared s3 bucket that
+  Nomad team developers can access. That commit SHA can be from any branch
+  that's pushed to remote. Ex. `nomad_sha =
+  "0b6b475e7da77fed25727ea9f01f155a58481b6c"`
+* `nomad_version`: provision this version from
+  [releases.hashicorp.com](https://releases.hashicorp.com/nomad). Ex. `nomad_version
+  = "0.10.2+ent"`
+
+## Profiles
+
+The `profile` field selects from a set of configuration files for Nomad,
+Consul, and Vault by uploading the files found in `./config/<profile>`. The
+profiles are as follows:
+
+* `full-cluster`: This profile is used for nightly E2E testing. It assumes at
+  least 3 servers and includes a unique config for each Nomad client.
+* `dev-cluster`: This profile is used for developer testing of a more limited
+  set of clients. It assumes at least 3 servers but uses the one config for
+  all the Linux Nomad clients and one config for all the Windows Nomad
+  clients.
+* `custom`: This profile is used for one-off developer testing of more complex
+  interactions between features. You can build your own custom profile by
+  writing config files to the `./config/custom` directory, which are protected
+  by `.gitignore`
+
+For each profile, application (Nomad, Consul, Vault), and agent type
+(`server`, `client_linux`, or `client_windows`), the agent gets the following
+configuration files, ignoring any that are missing.
+
+* `./config/<profile>/<application>/*`: base configurations shared between all
```

Comment:
I was a bit thrown off by not seeing `config/profile` changes but I see it was merged in https://github.com/hashicorp/nomad/pull/8828